server/controllers/offer-type/addOfferType.js

```diff
@@ -1,3 +1,28 @@
-module.exports = (req, res) => {
-  res.send(' add new offer type');
+const yup = require('yup');
+
+const { checkofferTypeName, addOfferType } = require('../../database/queries/offer-type');
+
+module.exports = (req, res, next) => {
+  const { name } = req.body;
+
+  const schema = yup.object({
+    name: yup.string().required(),
+  });
+  schema.validate({ name })
+    .then(() => {
+      checkofferTypeName(name)
+        .then((resultCheck) => {
+          if (!resultCheck.rowCount) {
+            addOfferType(name)
```

Comment:
@alaabadra I think if you added "return" here, like
```
return addOfferType(name)
  .then(( .......
```
Then any server errors from addOfferType will be passed up and out, and would be handled by your catch() on line 25, which would be nice.